design-proposals/project-deploymnet.md

```diff
@@ -0,0 +1,176 @@
+
+    clusterlink install --peer <name>
+
+This command installs the ClusterLink operator and creates the ClusterLink deployment.  
+Alternatively you can do it in steps:
+First, create and deploy the peer certificate:
+
+    clusterlink create peer --name <name>
+    kubectl create -f ```~/.config/clustrlink/<fabric_name>/<peer_nam>/secret.yaml```
+
+Second, create the ClustreLink operator:
+
+    kubectl create -f config/setup/
+
+Then, create clusterlink deployment:
+```
+kubectl apply -f - <<EOF
+apiVersion: clusterlink.net/v1alpha1
+kind: ClusterLink
+metadata:
+  namespace: clusterlink-ns
+  name: peer1
+spec:
+  dataplaneType: "envoy"
+  dataplaneReplicates: 1
+EOF
+```
+The operator will create:
+* ClusterLink controlplane deployment, including controlplane-pod, controlplane-service, and RBAC roles.  
+* ClusterLink dataplane deployment, including dataplane-pod and dataplane-service.
+* ClusterLink external access point using load-balancer/node-port/gateway-API service.
+  
+Overall, the ClusterLink deployment stages are:
+
+<img src="deployment.png" width="800" height="400" alt="ClusterLink Deployment Stages"/>    
+
+
+
+## Goals
+This design document should:
+
+- Define the steps for deploying a ClusterLink gateway to a K8s cluster.
+- Define the task functions of the ClusterLink operator.
+- Define the CRD (Custom Resource Definition) for the ClusterLink operator.
+
+## Non-Goals
+The following aspects are explicitly excluded and are out of scope in the current design:
+
+- This document focuses only on the deployment of ClusterLink to the K8s cluster. 
+The deployment to another environment as VMs is out of scope. 
+- The security aspects and user permissions concerning the creation of certificates, specifically for fabric or peer.
+## Proposal
+The ClusterLink deployment will be done in the following steps, (as described in more detail in the User section):
+
+1. Create a Fabric certificate if it does not exist.
+2. Create the ClusterLink operator.
+3. Deploy the ClusterLink CRD to the operator, which will create peer certificates, ClusterLink deployments (control plane and data plane), services, and external access point.
+
+### ClusterLink CRD
+The ClusterLink CRD includes the following fields:
+- Api version: clusterlink.net/v1alpha1
+- Kind: clusterlink
+
+- **Spec Section:**
+  
+    | Field name | Description | Default value |
+    | ---- | ----- | ------ |
+    |dataplaneType| Types of dataplane, supported values "go" or "envoy"|"envoy"|
+    |dataplaneReplicates| Number of dataplane replicas|1|
+    |logLevel| Log level severity| "info"|
+    |logFile| Path to a file where logs will be written| By default logs will be printed to stderr|
+    |image| The container registry to pull the project images. If empty, it will use the local registry| ghcr.io/clusterlink-net.
+    |serviceExposureType| Type of service to expose ClusterLink deployment, supported values: "load-balancer", "API-gateway", "node-port". |load-balancer|
```

Comment:
`exposureType`? 
Are there values that are needed based on the type of exposure? For example, do we want a configurable service name or use a hard coded (e.g.,) "load-balancer"/"node-port"?
The other end of the exposure should point to the control plane service, correct?